edx_rest_framework_extensions/permissions.py

```diff
@@ -1,9 +1,58 @@
 """ Permission classes. """
 from rest_framework.permissions import BasePermission
+from provider.oauth2.models import AccessToken as DOPAccessToken
+from edx_rest_framework_extensions.utils import jwt_decode_handler
+from oauth2_provider.ext.rest_framework.permissions import TokenHasScope
 
 
 class IsSuperuser(BasePermission):
     """ Allows access only to superusers. """
 
     def has_permission(self, request, view):
         return request.user and request.user.is_superuser
+
+
+class JWTRestrictedApplicationPermission(BasePermission):
```

Comment:
This might be better as `HasScopedToken`.

Reply:
Done.